### ISSUE TITLE
cmath.exp(1j) generates diagnostics for general type issue.

### ISSUE THREAD

**Issue by gyu-don** [microsoft/pyright#1679]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**

`cmath.exp(1j)` generates general type issues

```txt
Argument of type "complex" cannot be assigned to parameter "__z" of type "_C" in function "exp"
  Type "complex" cannot be assigned to type "_C"
    "__float__" is not present
    "__complex__" is not present
```

![Screenshot_20210323_183208](https://user-images.githubusercontent.com/13610313/112124781-141bf100-8c06-11eb-87ea-a82cc786dcd9.png)

**To Reproduce**
Write following codes on VS Code.

```py
import cmath
cmath.exp(1j)
```

**Expected behavior**
Don't generates this diagnostics because it's valid code.

**VS Code extension or command-line**
VS Code extension, Ver. 1.1.124

**Comment by gyu-don** [comment]
Oops, it seems typeshed's issues. `_C` is defined as 

```py
_C = Union[SupportsFloat, SupportsComplex]
```

in https://github.com/python/typeshed/blob/master/stdlib/cmath.pyi
but

```py
hasattr(1j, '__complex__') # => False
```
